src/PackagePublisher/Program.cs

```diff
@@ -0,0 +1,192 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.CommandLineUtils;
+using NuGet.Common;
+using NuGet.Packaging;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+
+namespace PackagePublisher
+{
+    public static class Program
+    {
+        private const string ApiKeyEnvironmentVariableName = "PACKAGEPUBLISHER_APIKey";
+        private static CommandOption _packagesDirectory;
+        private static CommandOption _feedToUploadTo;
+        private static CommandOption _maxRetryCountOption;
+
+        // Do NOT log the key
+        private static string _apiKey = null;
+        private static int _maxRetryCount = 10;
+
+        public static int Main(string[] args)
+        {
+            var app = new CommandLineApplication();
+            _packagesDirectory = app.Option(
+                "-d|--directory",
+                "The directory containing .nupkg file(s)",
+                CommandOptionType.SingleValue);
+            _feedToUploadTo = app.Option(
+                "-f|--feed",
+                "Feed to upload the .nupkg file(s) to",
+                CommandOptionType.SingleValue);
+            _maxRetryCountOption = app.Option(
+                "-c|--retryCount",
+                $"Max retry count to publish a package. Defaults to {_maxRetryCount}",
+                CommandOptionType.SingleValue);
+            var help = app.Option("-h|--help", "Show help", CommandOptionType.NoValue);
+
+            app.OnExecute(() =>
+            {
+                if (help.HasValue())
+                {
+                    app.ShowHelp();
```

Comment:
The commandlineapplication object has a standard way to show up the help that you would normally see when you do, let's say `dotnet restore --help`